src/pages/login/LoginVscodeCallback.tsx

```diff
@@ -13,10 +27,32 @@ const LoginVscodeCallback: React.FC = () => {
   const dispatch = useDispatch();
   const location = useLocation();
   const { t } = useTranslation('login');
-
+  const { isLoggedIn } = useSession();
+  const {
+    code,
+    provider: providerId,
+    'client-request-id': clientRequestId
+  } = parseQuery(location.search);
+  const isVscode = useTypedSelector(state => state.vscode.isVscode);
   const { access_token: accessToken, refresh_token: refreshToken } = parseQuery(location.search);
 
+  const launchVscode = () => {
+    window.location.href = `vscode://source-academy.source-academy/sso?code=${code}&client-request-id=${clientRequestId}`;
```

Comment:
Nit: Can use a value in `Constants.ts` for the base URL instead of being hard coded? But since unlike a frontend base URL, the extension is technically universal (deployment-agnostic), should be fine too. Just DRY if you need to do this in the future

Reply:
should it be a environment variable?

Reply:
I think no need? Do you foresee it to be configurable? There's only ever one extension (even when developing locally) right?

Reply:
Makes sense